src/components/FavoriteButton.component.tsx

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import { AiOutlineEye, AiOutlineEyeInvisible } from "react-icons/ai";
+import styled from "styled-components";
+
+import { defaultTheme } from "../theme";
+import { Button } from "./styled";
+
+interface Props {
+  fav: boolean;
+  onClick: () => void;
+  loading: boolean;
+}
+export const FavoriteButton = ({ fav, onClick, loading }: Props) => {
+  return (
+    <FavButton fav={fav} disabled={loading} onClick={() => onClick()}>
+      {loading ? (
+        "Loading..."
+      ) : !fav ? (
```

Comment:
please avoid using ternary operators this way, instead please try to keep it simple and more readable, you can create different variables for different ternary operators

Reply:
Hi Gerardo, I've just changed this logic to use if statements, it looks much better now thanks to you!